main.go

```diff
@@ -11,165 +13,126 @@ import (
+			config.DefaultConfig.TrustedBoot = true
+		}
 
-	if variant == "" {
-		// Set default variant
-		config.DefaultConfig.Variant = config.CoreVariant
-	} else {
 		// Try to load the variant
 		err := config.DefaultConfig.Variant.FromString(variant)
 		if err != nil {
-			fmt.Fprintf(os.Stderr, "%s\n", err.Error())
-			os.Exit(1)
+			return fmt.Errorf("error loading variant: %w", err)
 		}
-	}
 
-	if ksProvider == "" {
-		// Set default variant
-		config.DefaultConfig.KubernetesProvider = config.K3sProvider
-	} else {
-		// Try to load the variant
-		err := config.DefaultConfig.KubernetesProvider.FromString(ksProvider)
+		// Try to load the kubernetes provider
+		err = config.DefaultConfig.KubernetesProvider.FromString(ksProvider)
 		if err != nil {
-			fmt.Fprintf(os.Stderr, "%s\n", err.Error())
-			os.Exit(1)
+			return fmt.Errorf("error loading kubernetes provider: %w", err)
 		}
-	}
 
-	if config.DefaultConfig.KubernetesVersion == "latest" {
-		// Set default variant
-		config.DefaultConfig.KubernetesVersion = ""
-	}
+		if config.DefaultConfig.KubernetesVersion == "latest" {
+			config.DefaultConfig.KubernetesVersion = ""
+		}
 
-	logger := types.NewKairosLogger("kairos-init", config.DefaultConfig.Level, false)
-	logger.Infof("Starting kairos-init version %s", values.GetVersion())
-	logger.Debug(litter.Sdump(values.GetFullVersion()))
-
-	// Validate flags are being passed with actual values
-	// We dont care about variant and provider as we are setting that to a default value if value passed is empty
-	requiredFlags := []struct {
-		name  string
-		value string
-	}{
-		{"l", config.DefaultConfig.Level},
-		{"s", config.DefaultConfig.Stage},
-		{"m", config.DefaultConfig.Model},
-		{"r", config.DefaultConfig.Registry},
-		{"t", trusted},
-		{"f", config.DefaultConfig.FrameworkVersion},
-		{"version", version},
-	}
+		logger := types.NewKairosLogger("kairos-init", config.DefaultConfig.Level, false)
+		logger.Infof("Starting kairos-init version %s", values.GetVersion())
+		logger.Debug(litter.Sdump(values.GetFullVersion()))
 
-	for _, rf := range requiredFlags {
-		if rf.value == "" {
-			fmt.Fprintf(os.Stderr, "Error: %s flag is required to have a value\n", rf.name)
-			flag.Usage()
-			os.Exit(1)
+		// Parse the version number
+		sv, err := semver.NewSemver(version)
+		if err != nil {
+			return fmt.Errorf("error parsing version: %w", err)
 		}
-	}
 
-	// Parse the version number
-	sv, err := semver.NewSemver(version)
-	if err != nil {
-		fmt.Fprintf(os.Stderr, "Error: %s\n", err.Error())
-		flag.Usage()
-		os.Exit(1)
-	}
+		config.DefaultConfig.KairosVersion = *sv
+		litter.Config.HideZeroValues = true
+		litter.Config.HidePrivateFields = false
+		logger.Debug(litter.Sdump(config.DefaultConfig))
+
+		var runStages schema.YipConfig
 
-	config.DefaultConfig.KairosVersion = *sv
-	litter.Config.HideZeroValues = true
-	litter.Config.HidePrivateFields = false
-	logger.Debug(litter.Sdump(config.DefaultConfig))
+		if validate {
```

Comment:
now that we have cobra, maybe validate should be a different subcommand? It wasnt before because it was a mess manually, but maybe now it makes more sense?

Reply:
good idea. I was trying not to change the command's API to avoid breaking pipelines or trigger documentation changes. But let's do it and fix what breaks (if anything).

Reply:
If we extract it to a subcommand, this means callers of the original command will have to call it manually if they wanted validation to happen. Or I can explicitly call it here as well so we have both, the old way and a new sub-command that does only validation. What do you think?

Reply:
I went with the "keep both ways" solution.

Reply:
thats true, unless we can force cobra to trigger a subcommand or something if the old flag is used?

Maybe we can just add a deprecate notice in that flag, make that flag call the function for validation and add a subcommand for validation? So we maintain the flag for now but mark it as deprecate for the future while moving to the subcommand instead?

I dont care to be honest, but seems like validate as a flag was more of a workaround for not having a cli than anythng else, so the fastest w can fix it the better :D